### DISCUSSION TITLE
Chat history implementation (Langchain)

### DISCUSSION THREAD

**Discussion by ghost** [xtekky/gpt4free#1548] [Q&A]
Can someone put some light on this please. Is it possible to achieve chat history?  I just want to know if it's possible. Once I remember using langchain version .I was able to achieve chat history but it was a long time ago and I have not been active since last 6 months.

**Comment by hlohaus** [comment]
Hello, you can pass the history as additionally messages. The last message should be your prompt.

**Comment by TheGreatApollyon** [comment]
Can you share more about the langchain version you used?
